crates/block-producer/src/mempool/mod.rs

```diff
@@ -330,12 +336,9 @@ impl Mempool {
     ///
     /// # Panics
     ///
-    /// Panics if there is no block in flight or if the block number does not match the current
-    /// inflight block.
-    #[instrument(target = COMPONENT, skip_all, fields(block_number))]
-    pub fn block_failed(&mut self, block_number: BlockNumber) {
-        assert_eq!(block_number, self.chain_tip.child(), "Blocks must be submitted sequentially");
-
+    /// Panics if there is no block in flight.
+    #[instrument(target = COMPONENT, name = "mempool.block.rollback", skip_all)]
+    pub fn rollback_block(&mut self) {
```

Comment:
No longer takes `block_number` as this made state tracking more painful -- since the error case in the block builder wouldn't have direct access to the block number. We can revert this change, but I don't think it makes a difference at this stage.